terraform/fargate-examples/sqs-dynamic-target-tracking/main.tf

```diff
@@ -0,0 +1,641 @@
+# Lambda Function ECS scaling trigger
+################################################################################
+
+module "lambda_function_message_producer" {
+  source = "terraform-aws-modules/lambda/aws"
+
+  function_name      = "${local.name}-message-producer"
+  description        = "This function can be used to send test messages to the processing queue and trigger ASG scaling events."
+  handler            = "lambda_function.lambda_handler"
+  runtime            = "python3.9"
+  publish            = true
+  attach_policy_json = true
+  policy_json        = data.aws_iam_policy_document.lambda_role.json
+  source_path        = "../../../application-code/message-producer/"
+
+  environment_variables = {
+    queue_name                = module.processing_queue.this_sqs_queue_name
+    default_msg_proc_duration = local.default_msg_proc_duration
+    number_of_messages        = local.number_of_messages
+  }
+
+  allowed_triggers = {
+    PollSSMScale = {
+      principal  = "events.amazonaws.com"
+      source_arn = aws_cloudwatch_event_rule.sqs_message_producer.arn
+    }
+  }
+
+  cloudwatch_logs_retention_in_days = 30
+
+  tags = local.tags
+}
+
+module "lambda_function_target_bpi_update" {
+  source = "terraform-aws-modules/lambda/aws"
+
+  function_name      = "${local.name}-target_bpi_update"
+  description        = "This function regularly updates the target BPI of the ECS Service Target Tracking Policy"
+  handler            = "lambda_function.lambda_handler"
+  runtime            = "python3.9"
+  publish            = true
+  attach_policy_json = true
+  policy_json        = data.aws_iam_policy_document.lambda_role.json
+  source_path        = "../../../application-code/ecs-target-setter/"
+
+  environment_variables = {
+    scaling_policy_name       = local.scaling_policy_name
+    queue_name                = module.processing_queue.this_sqs_queue_name
+    app_metric_name           = local.app_metric_name
+    metric_type               = local.metric_type
+    metric_namespace          = local.metric_namespace
+    bpi_metric_name           = local.bpi_metric_name
+    default_msg_proc_duration = local.default_msg_proc_duration
+    desired_latency           = local.desired_latency
+  }
+
+  allowed_triggers = {
+    PollSSMScale = {
+      principal  = "events.amazonaws.com"
+      source_arn = aws_cloudwatch_event_rule.fargate_scaling.arn
+    }
+  }
+
+  cloudwatch_logs_retention_in_days = 30
+
+  tags = local.tags
+}
+
+################################################################################
+# Cloudwatch Events (EventBridge)
+################################################################################
+
+resource "aws_cloudwatch_event_rule" "fargate_scaling" {
+  name                = "ECSAutoscaleTargetBPIUpdate"
+  description         = "This rule is used for update ECS Autoscaling Target BPI"
+  schedule_expression = "rate(60 minutes)"
+
+  is_enabled = false
```

Comment:
is_enabled function is deprecated.

Reply:
ok sure. can you pls create change to   state = DISABLED and push a commit?